tests/contrib/test_gitlab.py

```diff
@@ -0,0 +1,93 @@
+from __future__ import unicode_literals
+
+import pytest
+import responses
+from urlobject import URLObject
+from flask import Flask
+from flask_dance.contrib.gitlab import make_gitlab_blueprint, gitlab
+from flask_dance.consumer import OAuth2ConsumerBlueprint
+from flask_dance.consumer.backend import MemoryBackend
+
+
+def test_blueprint_factory():
+    # Test with gitlab.com
+    glbp1 = make_gitlab_blueprint(
+        client_id="foo",
+        client_secret="bar",
+        scope="read_user",
+        redirect_to="index"
+    )
+    assert isinstance(glbp1, OAuth2ConsumerBlueprint)
+    assert glbp1.session.scope == "read_user"
+    assert glbp1.session.base_url == "https://gitlab.com/api/v4/"
+    assert glbp1.session.client_id == "foo"
+    assert glbp1.client_secret == "bar"
+    assert glbp1.authorization_url == "https://gitlab.com/oauth/authorize"
+    assert glbp1.token_url == "https://gitlab.com/oauth/token"
+    # Test with custom hostname
```

Comment:
Could you divide this into two separate test functions, please?

Reply:
will do.